rust/main/agents/validator/src/submit.rs

```diff
@@ -588,8 +588,8 @@ mod test {
             ) -> DbResult<Option<u64>>;
             fn store_highest_seen_message_nonce_number(&self, nonce: &u32) -> DbResult<()>;
             fn retrieve_highest_seen_message_nonce_number(&self) -> DbResult<Option<u32>>;
-            fn store_payload_id_by_message_id(&self, message_id: &H256, payload_id: &UniqueIdentifier) -> DbResult<()>;
-            fn retrieve_payload_id_by_message_id(&self, message_id: &H256) -> DbResult<Option<UniqueIdentifier>>;
+            fn store_payload_id_by_message_id(&self, message_id: &H256, payload_id: &UniqueIdentifier) -> DbResult<Vec<UniqueIdentifier>>;
```

Comment:
wondering why this would be helpful, the pattern for the `store` methods so far has been to return `()`

Reply:
User of all other `store_` methods has everything available to them at the time they invoke the method. It is not true `store_payload_id_by_message_id` method: user has only `message_id` and last `payment_id`. So, it can be useful to return the whole list of payment ids. And it does not cost much since no copy is happening there.

Reply:
Since payload ids will be known at the time when we store them, different return time is not needed. Fixed.